jiraworklog/src/Container/Timesheet/Timesheet.js

```diff
@@ -0,0 +1,276 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import User from '../../Component/User/User';
+import DateRangePicker from '@wojtekmaj/react-daterange-picker'
+import { Search, Grid, Button } from 'semantic-ui-react'
+import { escapeRegExp, filter, debounce } from 'lodash';
+import Api from '../../utility';
+import moment from 'moment';
+import './Timesheet.css';
+
+const date = new Date();
+
+class Timesheet extends Component {
+
+    state = {
+        users: [],
+        date: [new Date(date.getFullYear(), date.getMonth(), 1), new Date(date.getFullYear(), date.getMonth() + 1, 0)],
+        isLoading: false,
+        results: [],
+        value: '',
+        verticalsum: []
+    }
+    componentDidUpdate(props, PrevState) {
+        const { users, date } = this.state;
+        const [currentStateStartMonthDate, currentStateMonthEndDate] = date;
+        const [prevStateStartMonthDate, prevStateMonthEndDate] = PrevState.date;
+        if (currentStateStartMonthDate.toLocaleDateString() !== prevStateStartMonthDate.toLocaleDateString() || currentStateMonthEndDate.toLocaleDateString() !== prevStateMonthEndDate.toLocaleDateString()) {
+            this.setFiltedData(users);
+        }
+    }
+
+    getWLDatesArray = () => {
+        const { date } = this.state;
+        const dateArray = this.getDateArray(new Date(date[0]), new Date(date[1]));
+        const worklogArray = [];
+        dateArray.map(i => worklogArray.push(i.toLocaleDateString()));
```

Comment:
variable naming conventions? why are you still using dateArray? for example worklogArray => worklogs